### ISSUE TITLE
Error Initializing app: There was an error with the spawned command: npminstall

### ISSUE THREAD

**Issue by dantrevino** [driftyco/ionic-cli#1063]
Inonic@beta installed fine.  Error when trying to create apps.
![screenshot_2016-06-10-22-47-11](https://cloud.githubusercontent.com/assets/7738/15983512/b2d69764-2f5e-11e6-8c05-cc54cc115cd4.png)

**Comment by blackburnian** [comment]
I'm getting the same error.  Ionic@beta also installed without issue for me.

**Comment by jthoms1** [comment]
Could you try running the command with the verbose flag?  I have a feeling that everyone is receiving this same issue due to node-sass.

`ionic start cutePuppyPics --v2 --verbose`

**Comment by jthoms1** [comment]
Could you provide the output when using the --verbose flag.  The flag should allow you to see more details. Thanks!

**Comment by 3dd13** [comment]
there are few things you need to do to upgrade / update

1. upgrade nodejs

        brew upgrade node

2. upgrade npm

        npm install -g npm

3. set new registry

        npm set registry http://registry.npmjs.org

**Comment by blackburnian** [comment]
I tried updating node and npm with no success. Finally, I uninstalled node, npm, nvm, ionic, and cordova. Reinstalled nvm, and used that to reinstall node/npm, then reinstalled the others. This fixed the issue for me.

**Comment by jthoms1** [comment]
Closing this as a duplicate of #1013.

**Comment by zjw1918** [comment]
@blackburnian Thanks, that works.

**Comment by anilthummar** [comment]
Hello blackburnian,
i have installed below version for node,cordova and ionic 2
**C:\Users\Android>ionic -v
3.0.0-beta.5

C:\Users\Android>node -v
v6.10.0

C:\Users\Android>cordova -v
6.5.0**


I tried to creating ionic 2 project please help me ,
Error code:

**Downloading: https://github.com/driftyco/ionic2-app-base/archive/master.zip
Downloading: https://github.com/driftyco/ionic2-starter-sidemenu/archive/master.zip
Installing npm packages (may take a minute or two)...
Error with start undefined
Error Initializing app: There was an error with the spawned command: npminstall
There was an error with the spawned command: npminstall**

**Comment by blackburnian** [comment]
@anilthummar Did you try all the steps I mentioned?

> I tried updating node and npm with no success. Finally, I uninstalled node, npm, nvm, ionic, and cordova. Reinstalled nvm, and used that to reinstall node/npm, then reinstalled the others. This fixed the issue for me.

**Comment by anilthummar** [comment]
Hello Blackburnian,

I was try your steps , but still i am getting error to create ionic 2 project and i am using windows 10 OS .
-**can you please send me one example for ionic 2 sample code using VISUAL STUDIO 2015 tools.**

**Comment by blackburnian** [comment]
@anilthummar I don't have Visual Studio.

**Comment by anilthummar** [comment]
@blackburnian Thanks for the reply, then which tools to use for developing ionic project ??,

Please suggest me which one is better tools for developing ionic project in windows operating system.

**Comment by blackburnian** [comment]
@anilthummar I use WebStorm.  But it's really a matter of preference.  I don't use Windows.  I use Mac.  So not sure what is best for Windows.

**Comment by anilthummar** [comment]
@blackburnian 
Thank for the help.

**Comment by anilthummar** [comment]
@blackburnian 

I have successfully create ionic 2 project with blank template in windows 10, please follow below step.

**- C:\Users\Android>node -v**
      **v7.8.0**

**- C:\Users\Android>npm install -g cordova**

**- C:\Users\Android>npm install -g ionic**

**- C:\ionic>ionic start Project_Name blank --v2 --skip -npm**

**- C:\ionic>cd Project_Name**

**- C:\ionic\Project_Name>npm install -g cnpm --registry=https://registry.npm.taobao.org**

**- C:\ionic\Project_Name >ionic platform add android**

### To run or build your app for production, run
**- ionic run android --prod --release**

**Comment by anilthummar** [comment]
@blackburnian 

**I am still getting error When i am creating new ionic 2 project  Please help me .**

C:\ionic>ionic start cutePuppyPics2 --v2
Creating an Ionic 2.x app in C:\ionic\cutePuppyPics2 based on the tabs template.

**Downloading: https://github.com/driftyco/ionic2-app-base/archive/master.zip
Downloading: https://github.com/driftyco/ionic2-starter-tabs/archive/master.zip
Installing npm packages (may take a minute or two)...
Error with start undefined
Error Initializing app: There was an error with the spawned command: npminstall
There was an error with the spawned command: npminstall**

**Comment by anilthummar** [comment]
@jthoms1 
**I am still getting error When i am creating new ionic 2 project Please help me.**

C:\Windows\system32>**ionic -v
2.2.3**
C:\Windows\system32>**node -v
v6.10.2**

C:\Windows\system32>**cordova -v
6.5.0**

C:\Windows\system32>**ionic -info**|  **CLI v2.2.3**

**Error log:**

```
C:\IONIC2>ionic start cutePuppyPics --v2 --verbose
Task: title=start, name=start, summary=Starts a new Ionic project in the specified PATH, [options]=any flags for the command, <PATH>=directory for the new project, [template]=Starter templates can either come from a named template,
(ex: tabs, sidemenu, blank),
a Github repo, a Codepen url, or a local directory.
Codepen url, ex: http://codepen.io/ionic/pen/odqCz
Defaults to Ionic "tabs" starter template, --appname|-a=Human readable name for the app (Use quotes around the name), --id|-i=Package name for <widget id> config, ex: com.mycompany.myapp, title=Skip npm package installation, boolean=true, title=Create a basic structure without Cordova requirements, boolean=true, title=Setup the project to use Sass CSS precompiling, boolean=true, title=List starter templates available, boolean=true, --io-app-id=The Ionic.io app ID to use, --template|-t=Project starter template, boolean=true, title=Start an Ionic v1 project, --zip-file|-z=URL to download zipfile for starter template, isProjectTask=false, run=function run(ionic, argv) {
  if (argv.list || argv.l) {
    return templateUtils.listTemplates();
  }

  if (argv._.length < 2 && StartWizard) {
    return StartWizard.start();
  } else if(argv._length < 2) {
    return appLibUtils.fail('Invalid command', 'start');
  }

  if (argv._[1] === '.') {
    return log.error(chalk.red('Please name your Ionic project something meaningful other than \'.\''));
  }


  // Ensure that the folder name provided is a String
  // ie 5 becomes '5' AND '5' stays as '5'
  argv._[1] = argv._[1].toString();


  // This is temporary.... because I don't want to update app scripts before CLI v3 release
  // If defined as v1 in command line then remove the v2 flag
  // If no option or v2 option is passed then set the v2 flag
  argv.v2 = !argv.v1;

  var promptPromise;
  var options = appLibUtils.preprocessCliOptions(argv);
  var startingApp = true;

  // Grab the app's relative directory name
  if (fs.existsSync(options.targetPath)) {
    promptPromise = Start.promptForOverwrite(options.targetPath);
  } else {
    promptPromise = Q(true);
  }

  return promptPromise
  .then(function(promptToContinue) {
    if (!promptToContinue) {
      startingApp = false;
      log.info('\nIonic start cancelled by user.');
      return;
    }
    return Start.startApp(options);
  })
  .then(function() {
    if (startingApp) {
      return Start.printQuickHelp(options);
    }
  })
  .then(function() {
    if (startingApp) {
      return Start.promptLogin(options);
    }
  })
  .then(function() {
    if (options.v2 && startingApp) {
      //log.info('\nNew to Ionic? Get started here: http://ionicframework.com/getting-started\n');
    }
  })
  .catch(function(error) {
    log.error(error);
    throw error;
  });
}
Utils.preprocessCliOptions _=[start, cutePuppyPics], skip-npm=false, no-cordova=false, w=false, sass=false, s=false, list=false, l=false, v1=false, v2=true, verbose=true, $0=C:\Program Files\nodejs\node.exe C:\Users\Android\AppData\Roaming\npm\node_modules\ionic\bin\ionic
C:\IONIC2>ionic start cutePuppyPics --v2
Creating an Ionic 2.x app in C:\IONIC2\cutePuppyPics based on the tabs template.

Downloading: https://github.com/driftyco/ionic2-app-base/archive/master.zip
Downloading: https://github.com/driftyco/ionic2-starter-tabs/archive/master.zip
Installing npm packages (may take a minute or two)...
Error with start undefined
Error Initializing app: There was an error with the spawned command: npminstall
There was an error with the spawned command: npminstall.
```

**Comment by kuroikenshi** [comment]
I had the same issue. (Thanks to the Great Fire Wall... QAQ) 
I bought a proxy service, start shadowsocks. 

And modified "C:/Users/xxx/.npmrc":
```
registry=http://registry.npm.taobao.org/
strict-ssl=false
proxy=http://127.0.0.1:1080
https-proxy=http://127.0.0.1:1080
```

Then I cmd:
`ionic start MyIonicProject tutorial`

Succeed, but it took a long time(More than 10 Minutes) to see the message:
```
E:\ws\xxx>ionic start MyIonicProject tutorial
Creating an Ionic 2.x app in E:\ws\aionek\MyIonicProject based on the tutorial template.

Downloading: https://github.com/driftyco/ionic2-app-base/archive/master.zip
Downloading: https://github.com/driftyco/ionic2-starter-tutorial/archive/master.zip
Installing npm packages (may take a minute or two)...
\
♬ ♫ ♬ ♫  Your Ionic app is ready to go! ♬ ♫ ♬ ♫

Some helpful tips:

Run your app in the browser (great for initial development):
  ionic serve

Run on a device or simulator:
  ionic run ios[android,browser]

Share your app with testers, and test on device easily with the Ionic View companion app:
  http://view.ionic.io

E:\ws\xxx>
```

Give it a shot.

**Comment by anilthummar** [comment]
@kuroikenshi 

Please unblock amazon site in your system.
it will work charm.
